src/containers/my-cooperations/cooperation-notes/CooperationNotes.tsx

```diff
@@ -1,219 +1,170 @@
-    }
+  const deleteSelectedNote = (noteId: string) => {
+    return CooperationNotesService.deleteNote(id, noteId)
   }
 
+  const { mutate: deleteNote } = useMutation({
+    mutationFn: deleteSelectedNote,
+    queryKey: ['notes'],
+    onError: handleErrorAlert,
+    onSuccess: () => onSuccessResponse('delete')
+  })
+
   const onDeleteNote = (id: string) => {
     openDialog({
       message: 'cooperationsPage.modalMessages.confirmDeletionMessage',
-      sendConfirm: (isConfirmed: boolean) => void handleDelete(id, isConfirmed),
-      title: `cooperationsPage.modalMessages.confirmDeletionTitle`
+      title: `cooperationsPage.modalMessages.confirmDeletionTitle`,
+      sendConfirm: (isConfirmed: boolean) => {
+        if (isConfirmed) {
+          deleteNote(id)
+        }
+      }
     })
   }
 
-  const duplicateNote = useCallback(
-    (id?: string) => {
-      const note = notes.find((item) => item._id === id)
-      return createNoteService(note)
-    },
-    [notes, createNoteService]
-  )
-
-  const onDuplicateResponse = () => {
-    dispatch(
-      openAlert({
-        severity: snackbarVariants.success,
-        message: `cooperationsPage.modalMessages.successDuplication`
-      })
-    )
+  const updateSelectedNote = (params: {
+    noteId: string
+    data: CreateOrUpdateNoteParams
+  }) => {
+    return CooperationNotesService.updateNote(id, params.noteId, params.data)
   }
 
-  const { error: duplicationError, fetchData: duplicateItem } = useAxios({
-    service: duplicateNote,
-    fetchOnMount: false,
-    defaultResponse: null,
-    onResponseError,
-    onResponse: onDuplicateResponse
+  const { mutate: updateNote, isPending: updateNoteLoading } = useMutation({
+    mutationFn: updateSelectedNote,
+    queryKey: ['notes'],
+    onError: handleErrorAlert,
+    onSuccess: () => onSuccessResponse('update')
   })
 
-  const handleDuplicate = async (itemId: string) => {
-    await duplicateItem(itemId)
-    if (!duplicationError) await fetchData()
+  const duplicateNote = async (noteId: string) => {
+    const note = notes?.find((item) => item._id === noteId)
+    if (!note) {
+      return
+    }
+    return CooperationNotesService.createNote(note, id)
   }
 
-  const handleUpdate = async (data: CreateOrUpdateNoteParams) => {
-    await updateNote({ noteId: editableItemId, data })
+  const { mutate: duplicateItem } = useMutation({
+    mutationFn: duplicateNote,
+    queryKey: ['notes'],
+    onSuccess: () => onSuccessResponse('duplicate'),
+    onError: handleErrorAlert
+  })
+
+  const handleUpdate = (data: CreateOrUpdateNoteParams) => {
+    updateNote({ noteId: editableItemId, data })
     onCloseEdit()
-    if (!updateError) await fetchData()
   }
 
   const onCloseEdit = () => setEditableItemId('')
   const onCloseNote = () => setOpen(false)
   const onAddNoteOpen = () => setOpen(true)
 
-  const NotesList = notes.map((item: NoteResponse) =>
+  const NotesList = notes?.map((item: NoteResponse) =>
```

Comment:
Do not name variables with capital letters. Only **callable** React components should be named like that.
```suggestion
  const notesList = notes?.map((item: NoteResponse) =>
```